crates/slumber_core/src/template/render.rs

```diff
@@ -752,26 +752,56 @@ impl<'a> ChainTemplateSource<'a> {
             self.chain_id.to_string()
         };
 
-        let options = future::try_join_all(options.iter().enumerate().map(
-            |(i, template)| {
-                // Fork the local state, one copy for each new branch
-                let mut stack = stack.clone();
-                async move {
-                    template
-                        .render_chain_config(
-                            format!("options[{i}]"),
-                            context,
-                            &mut stack,
-                        )
-                        .await
-                }
-            },
-        ))
-        .await?;
+        let resolved_options: Vec<String> = match options {
+            SelectOptions::Fixed(options) => {
+                future::try_join_all(options.iter().enumerate().map(
+                    |(i, template)| {
+                        // Fork the local state, one copy for each new branch
+                        let mut stack = stack.clone();
+                        async move {
+                            template
+                                .render_chain_config(
+                                    format!("options[{i}]"),
+                                    context,
+                                    &mut stack,
+                                )
+                                .await
+                        }
+                    },
+                ))
+                .await?
+            }
+            SelectOptions::Dynamic(options) => {
+                let dynamic_options = options
+                    .render_chain_config("options", context, stack)
+                    .await?;
+
+                let raw_results: Vec<serde_json::Value> = serde_json::from_str(
+                    &dynamic_options,
+                )
+                .map_err(|error| ChainError::DynamicSelectOptions {
+                    error: Arc::new(error),
+                })?;
+
+                raw_results
```

Comment:
```suggestion
                // Convert all values to strings, for convenience. This may be a bit wonky for nested 
                // objects/arrays, but it should be obvious to the user what's going on so it's better 
                // than returning an error
                raw_results
```